GoogleForCreators/web-stories-wp
# Web Stories Embed Shortcode

## Summary

Allow for a shortcode to render an embedded story.

## Relevant Technical Choices

To make sure there lots of code was not repeated, some refactor was required. Much of the repeated logic was moved to parent class, called `Embed_Base` that could be extended by shortcode and block. 

Much of these changes relate to moving shortcode and block into there own namespaces. This mean all reference to the block class had to be updated. 

`Embed_base` into is called, so that html tag filter and script / styles are only registered once. 

I add a class option, so that embed block and shortcode have different classes for easier theme styling. 



## To-do



## User-facing changes



## Testing Instructions

1. Create a new post. 
2. Insert a shortcode block. 
3. Copy the following in block. 

Testing shortcode 
```
[web_stories_embed url="https://stories-new-wordpress-amp.pantheonsite.io/web-stories/13010/" height="300" width="200" title="Testing" align="right"]
```
4. Test the story renders. 

Transform test. 
1. Create a new post. 
2. Paste above snippet into editor. 
3. Shortcode should be transformed into embed story block.

---



Fixes #3135
